### PR TITLE
[Type Hints] `utils.remove_isolated_nodes`

### PR DESCRIPTION
**Pull request by LukeLIN-web**

### CODE REVIEW

File: torch_geometric/utils/isolated.py [hunk 1]
```diff
@@ -37,7 +37,9 @@ def contains_isolated_nodes(
     return torch.unique(edge_index.view(-1)).numel() < num_nodes
 
 
-def remove_isolated_nodes(edge_index, edge_attr=None, num_nodes=None):
+def remove_isolated_nodes(
+        edge_index: Tensor, edge_attr: Optional[Tensor] = None,
+        num_nodes: Optional[int] = None) -> Tuple[Tensor, Optional[Tensor], Tensor]:
```
Comment (EdisonLeeeee) [inline torch_geometric/utils/isolated.py]: Just wanted to add a comment but accidentally committed :-)

File: torch_geometric/utils/isolated.py [hunk 2]
```diff
@@ -37,7 +37,9 @@ def contains_isolated_nodes(
     return torch.unique(edge_index.view(-1)).numel() < num_nodes
 
 
-def remove_isolated_nodes(edge_index, edge_attr=None, num_nodes=None):
+def remove_isolated_nodes(
+        edge_index: Tensor, edge_attr: Optional[Tensor] = None,
```
Comment (EdisonLeeeee) [inline torch_geometric/utils/isolated.py]: Use `torch_geometric.typing.OptTensor` instead?

File: CHANGELOG.md [hunk 3]
```diff
@@ -37,6 +37,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added `BaseStorage.get()` functionality ([#5240](https://github.com/pyg-team/pytorch_geometric/pull/5240))
 - Added a test to confirm that `to_hetero` works with `SparseTensor` ([#5222](https://github.com/pyg-team/pytorch_geometric/pull/5222))
 ### Changed
+- Improved type hint support  ([#5659](https://github.com/pyg-team/pytorch_geometric/pull/5659))
```
[the remark this answers is not in the document]
Comment (rusty1s) [reply CHANGELOG.md:40]: The goal would be to group them all into one.

File: torch_geometric/utils/isolated.py [hunk 4]
```diff
@@ -37,7 +37,10 @@ def contains_isolated_nodes(
     return torch.unique(edge_index.view(-1)).numel() < num_nodes
 
 
-def remove_isolated_nodes(edge_index, edge_attr=None, num_nodes=None):
+def remove_isolated_nodes(
+    edge_index: Tensor, edge_attr: Optional[Tensor] = None,
+    num_nodes: Optional[int] = None
```
Comment (rusty1s) [inline torch_geometric/utils/isolated.py:42]:
```suggestion
    num_nodes: Optional[int] = None,
```
and let GitHub auto-format.

File: torch_geometric/utils/isolated.py [hunk 5]
```diff
@@ -37,7 +37,10 @@ def contains_isolated_nodes(
...
+) -> Tuple[Tensor, Optional[Tensor], Tensor]:
```
Comment (rusty1s) [inline torch_geometric/utils/isolated.py:43]: Can we change the rtype in the doc-string to Optional[Tensor] as well?

**Review by rusty1s — APPROVED** [review]
Thank you!

**Pull request merged**